libs/client-api/src/native/http_native.rs

```diff
@@ -122,24 +121,43 @@ impl Client {
   }
 
   async fn inner_refresh_token(&self) -> Result<(), AppResponseError> {
-    let retry_strategy = FixedInterval::new(Duration::from_secs(10)).take(4);
-    let action = RefreshTokenAction::new(self.token.clone(), self.gotrue_client.clone());
-    match RetryIf::spawn(retry_strategy, action, RefreshTokenRetryCondition).await {
-      Ok(_) => {
-        event!(tracing::Level::INFO, "refresh token success");
-        Ok(())
-      },
-      Err(err) => {
-        let err = AppError::from(err);
-        event!(tracing::Level::ERROR, "refresh token failed: {}", err);
+    let mut current_retry_count = 0;
+    let max_retry_count = 4;
 
-        // If the error is an OAuth error, unset the token.
-        if err.is_unauthorized() {
-          self.token.write().unset();
-        }
-        Err(err.into())
-      },
+    let refresh_token = self
+      .token
+      .read()
+      .as_ref()
+      .ok_or(GoTrueError::NotLoggedIn(
+        "fail to refresh user token".to_owned(),
+      ))?
+      .refresh_token
+      .as_str()
+      .to_owned();
+    let grant_req = Grant::RefreshToken(RefreshTokenGrant { refresh_token });
+
+    while current_retry_count < max_retry_count {
+      current_retry_count += 1;
+
+      match self.gotrue_client.token(&grant_req).await {
+        Ok(new_token) => {
+          self.token.write().set(new_token);
+          return Ok(());
+        },
+        Err(err) => {
+          warn!("refresh token failed: {}", err);
+          if err.is_network_error() {
+            std::thread::sleep(Duration::from_secs(2));
```

Comment:
You might want to use tokio::time::sleep instead of std::thread::sleep. They are different.

> std::thread::sleep
Synchronously blocks the current thread for the specified duration (in this case, 2 seconds).
During the sleep period, the thread is completely inactive and does no work.

> tokio::time::sleep
While the task is sleeping, the Tokio runtime can run other tasks on the same thread.

Reply:
got it